src/slack/messenger.ts

```diff
@@ -8,10 +8,22 @@ import {
   CurrentUser,
   Users,
   MessageReply,
-  Providers
+  Providers,
+  UserPresence
 } from "../types";
 import { SelfCommands } from "../constants";
 
+interface IDNDStatus {
+  dnd_enabled: boolean;
+  next_dnd_start_ts: number;
+  next_dnd_end_ts: number;
+  // The following fields are only available for the
+  // DND_UPDATED_SELF event.
+  snooze_enabled?: boolean;
```

Comment:
1. use these fields to auto-end a snooze session

2. for `dnd` periods, are we notified?